rfc-drafts/stream.md

```diff
@@ -221,31 +231,142 @@ existing code, perhaps as part of an edition migration.
 
 Designing such a migration feature is out of scope for this RFC.
 
-## IntoStream / FromStream traits, mirroring iterators
+## IntoStream / FromStream traits
+
+The `IntoStream` trait would provide a way to convert something into a `Stream`.
+
+The `FromStream` trait would provide a way to convert a `Stream` into something else.
+
+Implementing the `IntoStream` and `FromStream` traits on `Stream` is currently blocked on needing [`async fn` available in traits]( http://smallcultfollowing.com/babysteps/blog/2019/10/26/async-fn-in-traits-are-hard).
+
+These would need to provide similar iteration semantics as `Iterator`.
+* `for x in iter` uses `impl IntoIterator for T`
+* `for x in &iter` uses `impl IntoIterator for &T`
+* `for x in &mut iter` uses `impl IntoIterator for &mut T`
+
+The reasons not to do this right now are the same as the reasons for delaying convenience methods.
+
+## Other Traits
 
-* currently blocked on async fn in traits
-* The exact bounds are unclear.
-* the same as combinators
-* These would be needed to provide similar iteration semantics as Iterator:
-    * `for x in iter` uses `impl IntoIterator for T`
-    * `for x in &iter` uses `impl IntoIterator for &T`
-    * `for x in &mut iter` uses `impl IntoIterator for &mut T`
+Eventually, we will also want to add some (if not all) of the roster of traits we found useful for `Iterator`.
```

Comment:
I guess I would weaken this to "we may also want" -- presumably the reason for wanting these "families" of traits is to enable combinator methods, so I might emphasize that. e.g., if we want to have `reverse` on streams, then we need `DoubleEndedStream`, etc